src/Microsoft.TestPlatform.CommunicationUtilities/SocketClient.cs

```diff
@@ -35,7 +36,7 @@ protected SocketClient(Func<Stream, ICommunicationChannel> channelFactory)
             this.cancellation = new CancellationTokenSource();
             this.stopped = false;
 
-            this.tcpClient = new TcpClient();
+            this.tcpClient = new TcpClient { NoDelay = true };
```

Comment:
Using `NoDelay = true` can be a bit inefficient. 
More info here :
https://msdn.microsoft.com/en-us/library/system.net.sockets.tcpclient.nodelay(v=vs.110).aspx

Reply:
Yeah, As it also says we need for immediate/important communications. 

---
In reply to: [137197765](https://github.com/Microsoft/vstest/pull/1041#discussion_r137197765) [](ancestors = 137197765)